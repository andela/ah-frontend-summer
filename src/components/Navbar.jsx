/* eslint-disable react/no-unescaped-entities */
import React from "react";
import { connect } from 'react-redux';
import { NavLink } from 'react-router-dom';
import PropTypes from 'prop-types';
import { Menu, Button, Image, Icon } from "semantic-ui-react";
import "../styles/navbar.css";
import branding from '../assets/images/branding.png';

export const Navbar = (props) => {
    const { navBar } = props;
    return (
        <div>
            <Menu secondary className="mainMenu">
                <Menu.Item header>
                    <NavLink to="/">
                        <Image
                            src={branding}
                        />
                    </NavLink>
                </Menu.Item>
                <Menu.Menu position="right">
                    <Menu.Item name="menu" className="navItems">
                        <NavLink to="/">
                            Home
                        </NavLink>
                    </Menu.Item>
                    <Menu.Item name="tags" className="navItems">
                        <NavLink to="/">
                            Tags
                        </NavLink>
                    </Menu.Item>
                    {navBar.isLoggedIn ? (
                        <Menu.Menu position="right">
                            <NavLink to="/notifications">
                                <Icon size="large" name="bell outline" className="bellNotification" />
                            </NavLink>
                            <NavLink to="/profile">
                                <Icon size="big" name="user circle" className="avatar" />
                            </NavLink>
                        </Menu.Menu>
                    ) : (
                        <Menu.Menu position="right">
                            <Menu.Item name="signIn" className="navItems">
                                <NavLink to="/login">
                                    Sign in
                                </NavLink>
                            </Menu.Item>
                            <Menu.Item>
                                <NavLink to="/signup">
                                    <Button basic color="teal" className="getStartedBtn">
                                        Get Started
                                    </Button>
                                </NavLink>
                            </Menu.Item>
                        </Menu.Menu>
                    )}
                </Menu.Menu>
            </Menu>
        </div>
    );
};

const mapStateToProps = (state) => {
    return { navBar: state.navBar };
};

Navbar.propTypes = {
    navBar: PropTypes.object.isRequired
};

export default connect(mapStateToProps)(Navbar);
