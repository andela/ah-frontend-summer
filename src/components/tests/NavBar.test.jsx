import React from 'react';
import { shallow } from 'enzyme';
import { Navbar } from '../Navbar';
import navbarState from './fixtures/navBarState';

describe('Navbar test', () => {
    test('should render Navbar with default state', () => {
        const wrapper = shallow(<Navbar 
                                    navBar={navbarState[0]} 
                                    profile={{image: ""}} 
                                    fetchUserProfile={jest.fn()} />);
        expect(wrapper).toMatchSnapshot();
    });

    test('should render Navbar with logged in state', () => {
        const wrapper = shallow(<Navbar 
                                    navBar={navbarState[1]} 
                                    profile={{ image: "" }} 
                                    fetchUserProfile={jest.fn()} 
                                    />);
        expect(wrapper).toMatchSnapshot();
    });
});
