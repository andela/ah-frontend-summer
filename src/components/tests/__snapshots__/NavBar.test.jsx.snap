// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar test should render Navbar with default state 1`] = `
<div>
  <Menu
    className="mainMenu"
    secondary={true}
  >
    <MenuItem
      header={true}
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/"
      >
        <Image
          as="img"
          src="branding.png"
          ui={true}
        />
      </NavLink>
    </MenuItem>
    <MenuMenu
      position="right"
    >
      <MenuItem
        className="navHome"
        name="menu"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
        >
          Home
        </NavLink>
      </MenuItem>
      <MenuItem
        className="navTags"
        name="tags"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/tags"
        >
          Tags
        </NavLink>
      </MenuItem>
      <MenuMenu
        position="right"
      >
        <MenuItem
          className="navItems"
          name="signIn"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/login"
          >
            Sign in
          </NavLink>
        </MenuItem>
        <MenuItem>
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/signup"
          >
            <Button
              as="button"
              basic={true}
              className="getStartedBtn"
              color="teal"
            >
              Get Started
            </Button>
          </NavLink>
        </MenuItem>
      </MenuMenu>
    </MenuMenu>
  </Menu>
</div>
`;

exports[`Navbar test should render Navbar with logged in state 1`] = `
<div>
  <Menu
    className="mainMenu"
    secondary={true}
  >
    <MenuItem
      header={true}
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        to="/"
      >
        <Image
          as="img"
          src="branding.png"
          ui={true}
        />
      </NavLink>
    </MenuItem>
    <MenuMenu
      position="right"
    >
      <MenuItem
        className="navHome"
        name="menu"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/"
        >
          Home
        </NavLink>
      </MenuItem>
      <MenuItem
        className="navTags"
        name="tags"
      >
        <NavLink
          activeClassName="active"
          aria-current="page"
          to="/tags"
        >
          Tags
        </NavLink>
      </MenuItem>
      <MenuMenu
        position="right"
      >
        <MenuItem
          className="navItems"
          name="signIn"
        >
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/login"
          >
            Sign in
          </NavLink>
        </MenuItem>
        <MenuItem>
          <NavLink
            activeClassName="active"
            aria-current="page"
            to="/signup"
          >
            <Button
              as="button"
              basic={true}
              className="getStartedBtn"
              color="teal"
            >
              Get Started
            </Button>
          </NavLink>
        </MenuItem>
      </MenuMenu>
    </MenuMenu>
  </Menu>
</div>
`;
